diff -ruNp Git-Raw-0.90.orig/Makefile.PL Git-Raw-0.90/Makefile.PL
--- Git-Raw-0.90.orig/Makefile.PL	2022-10-27 10:50:15.000000000 +0200
+++ Git-Raw-0.90/Makefile.PL	2023-11-14 00:35:50.746645206 +0100
@@ -336,7 +336,8 @@ my %WriteMakefileArgs = (
     "Getopt::Long" => "2.35"
   },
   "DISTNAME" => "Git-Raw",
-  "INC" => "-I. -Ideps/libgit2 -I deps/libgit2/include -Ideps/libgit2/src/libgit2 -Ideps/libgit2/src/util -Ideps/libgit2/deps/http-parser -Ideps/libgit2/deps/zlib",
+  "INC" => "-isystem . -isystem /usr/include -isystem /usr/include/git2",
+  "LIBS" => "-lgit2",
   "LICENSE" => "perl",
   "MIN_PERL_VERSION" => "5.006",
   "NAME" => "Git::Raw",
